_posts/2022-02-23-#24098.md

```diff
@@ -10,14 +10,30 @@ status: upcoming
+
+As first [discussed in #17631](https://github.com/bitcoin/bitcoin/pull/17631#discussion_r733031180), the [current REST api](https://github.com/bitcoin/bitcoin/blob/master/doc/REST-interface.md) contains two endpoints `/headers/` and `/blockfilterheaders/` that rather unexpectedly use path parameters to control how many (filter) headers are returned in the response. While this is no critical issue, it is unintuitive and we are still early enough to easily phase this behaviour out and ensure new endpoints (if any) do not have to stick to non-standard behaviour just for internal consistency.
+
+In this PR, a new `HTTPRequest::GetQueryParameter` method is introduced to easily parse query parameters, as well as two new `/headers/` and `/blockfilterheaders/` endpoints that use a count query parameter are introduced. The old path parameter-based endpoints are kept without too much overhead, but the documentation now points to the new query parameter-based endpoints as the default interface to encourage standardness.
+
 ## Questions
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
+1. In HTTP requests, what is the difference between a `path` and a `query` parameter?
+1. What are the **benefits** of changing the `COUNT` parameter from a `path` parameter to a `query` parameter?
+1. What are the **drawbacks** of changing the `COUNT` parameter from a `path` parameter to a `query` parameter?
+1. *(general discussion)* If the effort of implementing a change is already done, what could be some good reasons for rejecting an (unharmful) code change anyway? Do you (dis)agree with [this comment](https://github.com/bitcoin/bitcoin/pull/24098#pullrequestreview-860377099)?
+1. Does this PR change any of the existing function signatures? If so, why? Can this cause any behaviour change?
+1. Can you list all the commits that introduce behaviour change(s)? Do you feel comfortable about these behaviour change(s)?
+1. Consider the request (signet) `GET /rest/blockfilterheaders/basic/2/0000004c6aad0c89c1c060e8e116dcd849e0554935cd78ff9c6a398abeac6eda.json?count=1`. What would the response be prior to this PR? What would the response be after this PR? Which (modified) function is responsible for this behaviour change? Try reasoning about it before verifying experimentally.
+1. One reviewer [raises](https://github.com/bitcoin/bitcoin/pull/24098#issuecomment-1027755825) the view that `/rest/headers/` is a collection endpoint instead of a single resource endpoint. Do you agree? If so, would you change the PR to implement this? What would be the drawbacks of doing that?
+
+### Bonus Questions
```

Comment:
These don't really need to be bonus questions, they seem like good normal questions to me! There's no expectations that we get through all of the questions during the meeting anyway